### DISCUSSION TITLE
In which file is constraint decoding implemented?

### DISCUSSION THREAD

**Discussion by estuday** [sgl-project/sglang#1150] [Q&A] [answered]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

Hi,
I would like to know in which file the constraint decoding is implemented. 
In outlines, constraint decoding is implemented by passing fsm into the logits_processers parameter of the generate method. I see that sgl also relies on outlines. Is the implementation logic the same?

### Reproduction

none

### Environment

none

**Comment by merrymercy** [comment, marked as answer]
https://github.com/sgl-project/sglang/tree/main/python/sglang/srt/constrained
